crates/bevy_app/src/app.rs

```diff
@@ -607,22 +623,29 @@ impl App {
     /// ```
     /// # use bevy_app::prelude::*;
     /// #
-    /// let app = App::empty().add_default_stages();
+    /// let app = App::empty().add_default_stages(AppThreading::Multi);
     /// ```
-    pub fn add_default_stages(&mut self) -> &mut Self {
-        self.add_stage(CoreStage::First, SystemStage::parallel())
+    pub fn add_default_stages(&mut self, threading: AppThreading) -> &mut Self {
+        self.add_stage(CoreStage::First, SystemStage::single_threaded())
             .add_stage(
                 StartupSchedule,
                 Schedule::default()
                     .with_run_criteria(ShouldRun::once)
-                    .with_stage(StartupStage::PreStartup, SystemStage::parallel())
-                    .with_stage(StartupStage::Startup, SystemStage::parallel())
-                    .with_stage(StartupStage::PostStartup, SystemStage::parallel()),
+                    .with_stage(StartupStage::PreStartup, Self::get_stage(threading))
+                    .with_stage(StartupStage::Startup, Self::get_stage(threading))
+                    .with_stage(StartupStage::PostStartup, Self::get_stage(threading)),
             )
-            .add_stage(CoreStage::PreUpdate, SystemStage::parallel())
-            .add_stage(CoreStage::Update, SystemStage::parallel())
-            .add_stage(CoreStage::PostUpdate, SystemStage::parallel())
-            .add_stage(CoreStage::Last, SystemStage::parallel())
+            .add_stage(CoreStage::PreUpdate, Self::get_stage(threading))
+            .add_stage(CoreStage::Update, Self::get_stage(threading))
+            .add_stage(CoreStage::PostUpdate, Self::get_stage(threading))
+            .add_stage(CoreStage::Last, Self::get_stage(threading))
+    }
+
+    fn get_stage(threading: AppThreading) -> SystemStage {
+        match threading {
+            AppThreading::Multi => SystemStage::parallel(),
+            AppThreading::Single => SystemStage::single_threaded(),
+        }
     }
```

Comment:
I think this should be a method on `AppThreading` instead of a static method. (Or simply `impl From<AppThreading> for SystemStage`)

Reply:
moved it to `AppThreading`. Implementing `From` felt a little weird to me.